app/src/main/java/com/onboarding/calculatorkotlin/mvp/model/CalculatorModel.kt

```diff
@@ -45,18 +49,83 @@ class CalculatorModel : CalculatorContract.Model {
     }
 
     override fun getLastModified(): String {
-        return if (secondOperand.isNotEmpty()) {
-            secondOperand
-        } else if (operator.isNotEmpty()) {
-            operator
-        } else {
-            firstOperand
+        return when {
+            secondOperand.isNotEmpty() -> {
+                secondOperand
+            }
+            operator.isNotEmpty() -> {
+                operator
+            }
+            else -> {
+                firstOperand
+            }
+        }
+    }
+
+    private fun operandEnabled(operand: String): Boolean {
+        return (operand != EMPTY_STRING && operand != SUB)
+    }
+
+    private fun operationEnabled(): Boolean {
+        var operationEnabled = true
+        if (getError() == Error.ERROR_DIVISION_BY_ZERO || getError() == Error.ERROR_INCOMPLETE_OPERATION) {
+            operationEnabled = false
+        }
+        return operationEnabled
+    }
+
+    override fun getError(): Error {
+        if (!operandEnabled(firstOperand) || !operandEnabled(secondOperand)) {
```

Comment:
here maybe you can use return if

Reply:
You can't because they are two separate ifs